<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>ForwardKinematics</class>
 <widget class="QMainWindow" name="ForwardKinematics">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>806</width>
    <height>507</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>ForwardKinematics</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QPushButton" name="pushButtonCalculate">
    <property name="geometry">
     <rect>
      <x>530</x>
      <y>230</y>
      <width>151</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Run Forward Kinematics</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButtonCancel">
    <property name="geometry">
     <rect>
      <x>690</x>
      <y>230</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Cancel</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="tableWidgetParameters">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>10</y>
      <width>751</width>
      <height>211</height>
     </rect>
    </property>
    <property name="minimumSize">
     <size>
      <width>631</width>
      <height>0</height>
     </size>
    </property>
    <property name="layoutDirection">
     <enum>Qt::LeftToRight</enum>
    </property>
    <property name="showDropIndicator" stdset="0">
     <bool>true</bool>
    </property>
    <property name="textElideMode">
     <enum>Qt::ElideRight</enum>
    </property>
    <attribute name="horizontalHeaderVisible">
     <bool>true</bool>
    </attribute>
    <attribute name="horizontalHeaderCascadingSectionResizes">
     <bool>false</bool>
    </attribute>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>100</number>
    </attribute>
    <attribute name="verticalHeaderVisible">
     <bool>false</bool>
    </attribute>
    <row>
     <property name="text">
      <string>1</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>2</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>3</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>4</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>5</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>6</string>
     </property>
    </row>
    <column>
     <property name="text">
      <string>i</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>αi−1 (degree)</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>ai−1 (in)</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>di (in)</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>θi (degree)</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Range of θi</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </column>
    <item row="0" column="0">
     <property name="text">
      <string>1</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="0" column="1">
     <property name="text">
      <string>0</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="0" column="2">
     <property name="text">
      <string>0.0</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="0" column="3">
     <property name="text">
      <string>0.0</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="0" column="4">
     <property name="text">
      <string>60</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="0" column="5">
     <property name="text">
      <string>-170 to 170</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="1" column="0">
     <property name="text">
      <string>2</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="1" column="1">
     <property name="text">
      <string>-90</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="1" column="2">
     <property name="text">
      <string>0.0</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="1" column="3">
     <property name="text">
      <string>0.0</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="1" column="4">
     <property name="text">
      <string>45</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="1" column="5">
     <property name="text">
      <string>-225 to 45</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="2" column="0">
     <property name="text">
      <string>3</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="2" column="1">
     <property name="text">
      <string>0</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="2" column="2">
     <property name="text">
      <string>17</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="2" column="3">
     <property name="text">
      <string>4.9</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="2" column="4">
     <property name="text">
      <string>25</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="2" column="5">
     <property name="text">
      <string>-250 to 75</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="3" column="0">
     <property name="text">
      <string>4</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="3" column="1">
     <property name="text">
      <string>-90</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="3" column="2">
     <property name="text">
      <string>0.8</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="3" column="3">
     <property name="text">
      <string>17.0</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="3" column="4">
     <property name="text">
      <string>15</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="3" column="5">
     <property name="text">
      <string>-135 to135</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="4" column="0">
     <property name="text">
      <string>5</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="4" column="1">
     <property name="text">
      <string>90</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="4" column="2">
     <property name="text">
      <string>0.0</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="4" column="3">
     <property name="text">
      <string>0.0</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="4" column="4">
     <property name="text">
      <string>-10</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="4" column="5">
     <property name="text">
      <string>-100 to 100</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="5" column="0">
     <property name="text">
      <string>6</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="5" column="1">
     <property name="text">
      <string>-90</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="5" column="2">
     <property name="text">
      <string>0.0</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="5" column="3">
     <property name="text">
      <string>0.0</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="5" column="4">
     <property name="text">
      <string>-30</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
    <item row="5" column="5">
     <property name="text">
      <string>-180 to 180</string>
     </property>
     <property name="textAlignment">
      <set>AlignHCenter|AlignVCenter|AlignCenter</set>
     </property>
    </item>
   </widget>
   <widget class="QTableWidget" name="tableWidgetLinkTransMatrix">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>300</y>
      <width>261</width>
      <height>151</height>
     </rect>
    </property>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>60</number>
    </attribute>
    <row>
     <property name="text">
      <string>X</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>Y</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>Z</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string/>
     </property>
    </row>
    <column>
     <property name="text">
      <string>X'</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Y'</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Z'</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>P'</string>
     </property>
    </column>
    <item row="0" column="0">
     <property name="text">
      <string/>
     </property>
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="0" column="1">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="0" column="2">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="0" column="3">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="1" column="0">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="1" column="1">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="1" column="2">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="1" column="3">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="2" column="0">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="2" column="1">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="2" column="2">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="2" column="3">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="3" column="0">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="3" column="1">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="3" column="2">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="3" column="3">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
   </widget>
   <widget class="QPushButton" name="pushButtonLinkTransformationMatrix">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>270</y>
      <width>91</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Calculate i-1T i</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox">
    <property name="geometry">
     <rect>
      <x>80</x>
      <y>270</y>
      <width>42</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>6</number>
    </property>
   </widget>
   <widget class="QTableWidget" name="tableWidgetManipulatorMatrix">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>300</y>
      <width>261</width>
      <height>151</height>
     </rect>
    </property>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>60</number>
    </attribute>
    <row>
     <property name="text">
      <string>X</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>Y</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>Z</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string/>
     </property>
    </row>
    <column>
     <property name="text">
      <string>X'</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Y'</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>Z'</string>
     </property>
    </column>
    <column>
     <property name="text">
      <string>P'</string>
     </property>
    </column>
    <item row="0" column="0">
     <property name="text">
      <string/>
     </property>
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="0" column="1">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="0" column="2">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="0" column="3">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="1" column="0">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="1" column="1">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="1" column="2">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="1" column="3">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="2" column="0">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="2" column="1">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="2" column="2">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="2" column="3">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="3" column="0">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="3" column="1">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="3" column="2">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="3" column="3">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
   </widget>
   <widget class="Line" name="line">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>250</y>
      <width>751</width>
      <height>20</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>270</y>
      <width>71</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Link Number</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>290</x>
      <y>270</y>
      <width>181</width>
      <height>21</height>
     </rect>
    </property>
    <property name="text">
     <string>Manipulator Transformation Matrix</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="tableWidgetEndEffector">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>300</y>
      <width>81</width>
      <height>151</height>
     </rect>
    </property>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>60</number>
    </attribute>
    <row>
     <property name="text">
      <string>X</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>Y</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>Z</string>
     </property>
    </row>
    <column>
     <property name="text">
      <string/>
     </property>
    </column>
    <item row="0" column="0">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="1" column="0">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="2" column="0">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>270</y>
      <width>101</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>End Effector Position</string>
    </property>
   </widget>
   <widget class="QTableWidget" name="tableWidgetAngles">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>300</y>
      <width>91</width>
      <height>151</height>
     </rect>
    </property>
    <attribute name="horizontalHeaderDefaultSectionSize">
     <number>60</number>
    </attribute>
    <row>
     <property name="text">
      <string>γ</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>β</string>
     </property>
    </row>
    <row>
     <property name="text">
      <string>α</string>
     </property>
    </row>
    <column>
     <property name="text">
      <string/>
     </property>
    </column>
    <item row="0" column="0">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="1" column="0">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
    <item row="2" column="0">
     <property name="flags">
      <set>ItemIsSelectable|ItemIsDragEnabled|ItemIsEnabled</set>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>680</x>
      <y>270</y>
      <width>121</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string>Fixed Oriendation Angles</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>806</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuPUMA">
    <property name="title">
     <string>PUMA</string>
    </property>
   </widget>
   <addaction name="menuPUMA"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
